sql/migrations/20180523190341_world.sql

```diff
@@ -0,0 +1,344 @@
+('56855', '11673', '409', '0', '0', '669.2400', '-532.061', '-210.223', '0.215849', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56856', '11673', '409', '0', '0', '708.1200', '-604.032', '-209.670', '4.374480', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56857', '11673', '409', '0', '0', '729.6020', '-485.480', '-212.510', '4.538390', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56858', '11673', '409', '0', '0', '709.8370', '-932.331', '-191.297', '1.980750', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56859', '11673', '409', '0', '0', '705.4090', '-1197.96', '-119.933', '2.581690', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56860', '11673', '409', '0', '0', '576.6950', '-777.409', '-207.321', '4.422850', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56861', '11673', '409', '0', '0', '814.4380', '-973.453', '-208.881', '5.024870', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56862', '11673', '409', '0', '0', '607.6860', '-1160.48', '-198.044', '1.193060', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56863', '11673', '409', '0', '0', '659.5690', '-1071.07', '-190.902', '6.172250', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56864', '11673', '409', '0', '0', '746.8200', '-997.854', '-177.562', '4.278700', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10'),
+('56865', '11673', '409', '0', '0', '796.0060', '-639.668', '-204.321', '0.340817', '1080', '1080', '0', '0', '152352', '0', '0', '2', '1', '0', '0', '10');
+
+REPLACE INTO `creature_template_addon` (`entry`, `patch`, `mount`, `bytes1`, `b2_0_sheath`, `b2_1_flags`, `emote`, `moveflags`, `auras`) VALUES
+('11673', '0', '0', '0', '1', '16', '0', '0', '18950');
+
+SET @CGUID := 56851;
+SET @POINT := 0;
+DELETE FROM `creature_movement` WHERE `id`=@CGUID;
+INSERT INTO `creature_movement` (`id`, `point`, `position_x`, `position_y`, `position_z`, `waittime`, `script_id`, `orientation`) VALUES
+(@CGUID, @POINT := @POINT + 1, '1144.05', '-662.684', '-131.463', '0', '0', '100'),
+(@CGUID, @POINT := @POINT + 1, '1160.29', '-696.880', '-132.486', '0', '0', '100'),
+(@CGUID, @POINT := @POINT + 1, '1129.03', '-721.049', '-139.580', '0', '0', '100'),
+(@CGUID, @POINT := @POINT + 1, '1118.51', '-712.138', '-144.258', '0', '0', '100'),
+(@CGUID, @POINT := @POINT + 1, '1104.69', '-722.130', '-146.983', '0', '0', '100'),
+(@CGUID, @POINT := @POINT + 1, '1090.82', '-731.363', '-149.987', '0', '0', '100'),
+(@CGUID, @POINT := @POINT + 1, '1070.79', '-717.562', '-153.483', '0', '0', '100'),
+(@CGUID, @POINT := @POINT + 1, '1080.75', '-681.572', '-159.873', '0', '0', '100'), -- random movement for ~2 mins, not possible in mangos
```

Comment:
Its possible. Assign a script id with these 2 actions:

ACTION 1
Delay: 0
Command: SCRIPT_COMMAND_MOVEMENT (20)
Datalong: RANDOM_MOTION_TYPE (1)

ACTION 2
Delay: 120
Command: SCRIPT_COMMAND_MOVEMENT (20)
Datalong: WAYPOINT_MOTION_TYPE (2)
Datalong2: 1 (True)
Datalong3: 9 (the starting point)

Reply:
Is that not going to happen around its spawn location, rather than waypoint 8. I see no parameters to control this.

Reply:
The `script_id` column in the `creature_movement` table is a script from `creature_movement_scripts` that will be started upon reaching this point. What you want to do is have it switch to random movement immediately upon reaching point 8, and then start waypoint movement again from the next point (9) after 120 seconds.

This should do it:
```
DELETE FROM `creature_movement_scripts` WHERE `id`=56851;
INSERT INTO `creature_movement_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES (56851, 0, 20, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Ancient Core Hound - Start Random Movement');
INSERT INTO `creature_movement_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES (56851, 0, 20, 2, 1, 9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Ancient Core Hound - Resume Waypoint Movement from Point 9');
```

Reply:
Not for me, the random movement is performed around the creature's spawn location, that's my issue.

So what it does is: Spawn -> WP1 ... WP8 -> Spawn -> Random -> WP9

That's the issue we also had in mangos, if I recall correctly. it's been a while though.

Reply:
Ah, what about setting the creature's home location to the current location before starting random movement? I think that may fix it.

Reply:
possibly, but at some point it gets ugly when you have to reset it when the creature dies or reaches other wps

maybe the cleaner solution is to upgrade the command and add some basic parameters, wp or x, y, z, distance ...

Reply:
Seems like we also need a `spawndist` override parameter for MotionMaster::MoveRandom, for creatures where random movement is not the default.

Reply:
I made the needed changes to MoveRandom.
https://github.com/lh-server/core/pull/103

Here is the script:
```
DELETE FROM `creature_movement_scripts` WHERE `id`=56851;
INSERT INTO `creature_movement_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES (56851, 0, 20, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Ancient Core Hound - Set Movement to Idle');
INSERT INTO `creature_movement_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES (56851, 3, 20, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 'Ancient Core Hound - Start Random Movement');
INSERT INTO `creature_movement_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES (56851, 120, 20, 2, 1, 9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Ancient Core Hound - Resume Waypoint Movement from Point 9');
```

You must also set a `waittime` for point 8, it can be just a second, so that he stops.

Reply:
Looks good to me, I'll update when this is merged. Nice!